### PR TITLE
Add List Price Page

### PR DESCRIPTION
**Pull request by RahulJha-mastek**
Story AB#14656
Task AB#15172

### CODE REVIEW

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/MappingProfiles/SolutionDetailsProfile.cs [hunk 1]
```diff
@@ -160,16 +163,40 @@ public SolutionDetailsProfile()
...
+                {
+                    dest.Price = src.Price == null ? null : Math.Round(src.Price.Value, 2);
+                })
+                .ForMember(dest => dest.CurrencyCode, opt => opt.MapFrom(src => src.CurrencyCode == null ? null : CurrencyCodeSigns.Code[src.CurrencyCode]))
+                .ForMember(dest => dest.Price, opt => opt.Ignore())
+                .ForMember(dest => dest.Unit, opt => opt.MapFrom(src => string.Concat(
+                    src.PricingUnit == null ? string.Empty : src.PricingUnit.Description, 
+                    " ",
+                    src.TimeUnit == null ? string.Empty : src.TimeUnit.Description)));
+
```
Comment (kiran-mastek) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/MappingProfiles/SolutionDetailsProfile.cs:191]: Should cater for condition of `src.CataloguePrices` being null here

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/MappingProfiles/SolutionDetailsProfile.cs [hunk 2]
```diff
@@ -160,16 +163,40 @@ public SolutionDetailsProfile()
...
                 .ForMember(dest => dest.PublicCloud, opt => opt.Ignore())
                 .ForMember(dest => dest.PrivateCloud, opt => opt.Ignore())
                 .ForMember(dest => dest.HybridHostingType, opt => opt.Ignore())
                 .ForMember(dest => dest.OnPremise, opt => opt.Ignore())
                 .IncludeBase<CatalogueItem, SolutionDisplayBaseModel>();
+
+            CreateMap<CataloguePrice, PriceViewModel>()
+                .BeforeMap(
+                (src, dest) =>
+                {
```
Comment (kiran-mastek) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/MappingProfiles/SolutionDetailsProfile.cs:182]: This mapping should be possible as normal - shouldn't need to do it in `BeforeMap`

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/MappingProfiles/SolutionDetailsProfile.cs [hunk 3]
```diff
@@ -160,16 +163,40 @@ public SolutionDetailsProfile()
...
+                    dest.Price = src.Price == null ? null : Math.Round(src.Price.Value, 2);
+                })
+                .ForMember(dest => dest.CurrencyCode, opt => opt.MapFrom(src => src.CurrencyCode == null ? null : CurrencyCodeSigns.Code[src.CurrencyCode]))
+                .ForMember(dest => dest.Price, opt => opt.Ignore())
```
Comment (kiran-mastek) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/MappingProfiles/SolutionDetailsProfile.cs:186]: The below should look neater with string interpolation:
`$"{(src.PricingUnit == null ? string.Empty : src.PricingUnit.Description)} {(src.TimeUnit == null ? string.Empty : src.TimeUnit.Description)}"`

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/MappingProfiles/SolutionDetailsProfile.cs (hunk 1 above)
Comment (RahulJha-mastek) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/MappingProfiles/SolutionDetailsProfile.cs:188]: Added Null check PreCondition for `src.CataloguePrices`

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/MappingProfiles/SolutionDetailsProfile.cs (hunk 3 above)
Comment (RahulJha-mastek) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/MappingProfiles/SolutionDetailsProfile.cs]: Updated with `string interpolation`

**Review by PhysicsJon — APPROVED** [review]

**Pull request merged**